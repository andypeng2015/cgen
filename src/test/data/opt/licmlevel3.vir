module foo

export function w $foo(w %x) {
@start:
  %i = copy.w 0_w
  %z = copy.w 0_w
  jmp @loop1
@loop1:
  %c = lt.w %i, 10_w
  br %c, @body1, @done
@body1:
  %j = copy.w 0_w
  jmp @loop2
@loop2:
  %c = lt.w %j, 10_w
  br %c, @body2, @done2
@body2:
  %k = copy.w 0_w
  jmp @loop3
@loop3:
  %c = lt.w %k, 10_w
  br %c, @body3, @done3
@body3:
  %y = add.w %x, 1_w
  %z = add.w %z, %y
  %k = add.w %k, 1_w
  jmp @loop3
@done3:
  %j = add.w %j, 1_w
  jmp @loop2
@done2:
  %i = add.w %i, 1_w
  jmp @loop1
@done:
  ret %z
}
