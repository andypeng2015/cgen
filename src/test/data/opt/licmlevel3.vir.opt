module foo

export function w $foo(w %x) {
@2:
  %8 = add.w %x, 0x1_w ; @32
  jmp @3(0x0_w, 0x0_w)
@3(%z.2, %i.2):
  %9 = lt.w %i.2, 0xa_w ; @33
  br %9, @9(%z.2, 0x0_w), @7
@9(%z.3, %j.2):
  %10 = lt.w %j.2, 0xa_w ; @34
  br %10, @14(%z.3, 0x0_w), @12
@14(%z.4, %k.2):
  %11 = lt.w %k.2, 0xa_w ; @35
  br %11, @16, @17
@16:
  %13 = add.w %z.4, %8 ; @37
  %14 = add.w %k.2, 0x1_w ; @38
  jmp @14(%13, %14)
@17:
  %12 = add.w %j.2, 0x1_w ; @36
  jmp @9(%z.4, %12)
@12:
  %15 = add.w %i.2, 0x1_w ; @39
  jmp @3(%z.3, %15)
@7:
  ret %z.2
}
