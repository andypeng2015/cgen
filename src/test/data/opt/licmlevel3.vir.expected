module foo

export function w $foo(w %x) {
@0:
  %0 = add.w %x, 0x1_w ; @22
  jmp @1(0x0_w, 0x0_w)
@1(%z.2, %i.2):
  %1 = lt.w %i.2, 0xa_w ; @23
  br %1, @7(%z.2, 0x0_w), @5
@7(%z.3, %j.2):
  %2 = lt.w %j.2, 0xa_w ; @24
  br %2, @12(%z.3, 0x0_w), @10
@12(%z.4, %k.2):
  %3 = lt.w %k.2, 0xa_w ; @25
  br %3, @14, @15
@14:
  %5 = add.w %z.4, %0 ; @27
  %6 = add.w %k.2, 0x1_w ; @28
  jmp @12(%5, %6)
@15:
  %4 = add.w %j.2, 0x1_w ; @26
  jmp @7(%z.4, %4)
@10:
  %7 = add.w %i.2, 0x1_w ; @29
  jmp @1(%z.3, %7)
@5:
  ret %z.2
}
